libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationTelemetry.java

```diff
@@ -213,8 +213,9 @@ public void updateLocationEngineNameAndSimulation(LocationEngine locationEngine)
     }
   }
 
-  public void updateLocation(Location location) {
-    gpsEventFactory.gpsReceived();
+  public void updateLocation(Context context, Location location) {
+    NavigationPerformanceMetadata metadata = new MetadataBuilder().constructMetadata(context);
```

Comment:
Now we're passing `metadata` to all the events (including `InitialGpsEvent`), as this is executed every location update, would that have any performance impact? Could we generate `NavigationPerformanceMetadata` once? It seems it could be a singleton because metadata is constant.

Reply:
Yeah, you are right! Thanks for vigilance :suspect: